lib/app.rb

```diff
@@ -0,0 +1,32 @@
+require_relative './game.rb'
+require_relative './view.rb'
+require_relative './player.rb'
+require_relative './input_handler.rb'
+require_relative './constants.rb'
+require_relative './parser.rb'
+
+class App
+    def initialize(args={})
+        @game = args.fetch(:game, new_game)
+        @view = args.fetch(:view, View.new)
+        @parser = args.fetch(:parser, Parser.new)
+        @input_reader = args.fetch(:input_reader, InputHandler.new(IO.new(1)))
+    end
```

Comment:
Give [this](http://misko.hevery.com/code-reviewers-guide/flaw-constructor-does-real-work/) a read.